examples/2d/jeans_instability.jl

```diff
@@ -0,0 +1,100 @@
+# combining both semidiscretizations for Euler + self-gravity
+parameters = ParametersEulerGravity(background_density=1.5e7, # aka rho0
+                                    gravitational_constant=6.674e-8, # aka G
+                                    cfl=2.4,
+                                    n_iterations_max=1000,
+                                    timestep_gravity=timestep_gravity_erk52_3Sstar!)
+
+semi = SemidiscretizationEulerGravity(semi_euler, semi_gravity, parameters)
+
+
+###############################################################################
+# ODE solvers, callbacks etc.
+tspan = (0.0, 5.0)
+ode = semidiscretize(semi, tspan);
+
+summary_callback = SummaryCallback()
+
+analysis_interval = 100
+alive_callback = AliveCallback(analysis_interval=analysis_interval)
+
+Trixi.pretty_form_repl(::Val{:energy_potential}) = "∑e_potential"
+Trixi.pretty_form_file(::Val{:energy_potential}) = "e_potential"
+
+function Trixi.analyze(::Val{:energy_potential}, du, u_euler, t, semi::SemidiscretizationEulerGravity)
+
+  u_gravity = Trixi.wrap_array(semi.cache.u, semi.semi_gravity)
+
+  mesh, equations_euler, dg, cache = Trixi.mesh_equations_solver_cache(semi.semi_euler)
+  _, equations_gravity, _, _ = Trixi.mesh_equations_solver_cache(semi.semi_gravity)
+
+  e_potential = Trixi.integrate(mesh, equations_euler, dg, cache, u_euler, equations_gravity, u_gravity) do u, i, j, element, equations_euler, dg, equations_gravity, u_gravity
+    u_euler_local   = Trixi.get_node_vars(u_euler,   equations_euler,   dg, i, j, element)
+    u_gravity_local = Trixi.get_node_vars(u_gravity, equations_gravity, dg, i, j, element)
+    # OBS! subtraction is specific to Jeans instability test where rho0 = 1.5e7
+    return (u_euler_local[1] - 1.5e7) * u_gravity_local[1]
+  end
+  return e_potential
+end
+
+analysis_callback = AnalysisCallback(semi_euler, analysis_interval=analysis_interval,
+                                     save_analysis=true,
+                                     extra_analysis_integrals=(entropy, energy_total, energy_kinetic, energy_internal, Val(:energy_potential)))
+
+save_solution = SaveSolutionCallback(solution_interval=10,
+                                     save_initial_solution=true,
+                                     save_final_solution=true,
+                                     solution_variables=:primitive)
+
+# TODO: Taal, IO
+# restart_interval = 10
+
+stepsize_callback = StepsizeCallback(cfl=1.0)
+
+callbacks = CallbackSet(summary_callback, stepsize_callback, analysis_callback, save_solution, alive_callback)
+
+
+###############################################################################
+# run the simulation
+sol = solve(ode, CarpenterKennedy2N54(williamson_condition=false), dt=stepsize_callback(ode),
+            save_everystep=false, callback=callbacks);
+
+println("Number of gravity subcycles: ", semi.gravity_counter.ncalls_since_readout)
+
+nothing
```

Comment:
I've seen you've gone back to omitting the `return`... at the risk of bikeshedding... I'd be open to revise our policy about that, i.e., to allow implicit returns (at least if there is only a single exit from a function and if the return statement is the last line and separated by a blank line maybe), but we should probably mention it at one of the next meetings.

Reply:
I haven't been consistent here, sometimes I used `return`, sometimes not. We should discuss this again and go over the code.